smarts/core/chassis.py

```diff
@@ -509,9 +509,11 @@ def step(self, current_simulation_time):
 
     def inherit_physical_values(self, other: BoxChassis):
         self.set_pose(other.pose)
-        # bullet wants m/s
-        velocity = radians_to_vec(self.pose.heading) * other.speed
-        self.speed = other.speed
+        self._initialize_speed(other.speed)
+
+    def _initialize_speed(self, speed: float):
+        self.speed = speed
+        velocity = radians_to_vec(self.pose.heading) * speed
         self._client.resetBaseVelocity(self._bullet_id, [*velocity, 0])
```

Comment:
This is nice. We can also enhance the performance by reseting the rotational velocities of the wheels if it has some lags.